sdk/ai/azure-ai-inference/src/main/java/com/azure/ai/inference/ChatCompletionClientTracer.java

```diff
@@ -0,0 +1,648 @@
+    private static final String FINISH_REASON_CANCELLED = "{\"finish_reason\": \"cancelled\"}";
+    private static final StartSpanOptions START_SPAN_OPTIONS = new StartSpanOptions(SpanKind.CLIENT);
+
+    private static final ConfigurationProperty<Boolean> CAPTURE_MESSAGE_CONTENT
+        = ConfigurationPropertyBuilder.ofBoolean("otel.instrumentation.genai.capture_message_content")
+            .environmentVariableName(Configuration.PROPERTY_AZURE_TRACING_DISABLED)
+            .shared(true)
+            .defaultValue(false)
+            .build();
+    private static final Configuration GLOBAL_CONFIG = Configuration.getGlobalConfiguration();
+
+    private final ClientLogger logger;
+    private final String host;
+    private final int port;
+    private final boolean captureContent;
+    private final Tracer tracer;
+
+    //<editor-fold desc="operation contracts">
+    /**
+     * Reference to the operation performing the actual completion call.
+     */
+    @FunctionalInterface
+    public interface SyncCompleteOperation {
+        /**
+         * invokes the operation.
+         *
+         * @param completeRequest The completeRequest parameter for the {@code operation}.
+         * @param requestOptions The requestOptions parameter for the {@code operation}.
+         * @return chat completions for the provided chat messages.
+         */
+        ChatCompletions invoke(BinaryData completeRequest, RequestOptions requestOptions);
+    }
+
+    /**
+     * Reference to the async operation performing the actual completion call.
+     */
+    @FunctionalInterface
+    public interface CompleteOperation {
+        /**
+         * invokes the operation.
+         *
+         * @param completeRequest The completeRequest parameter for the {@code operation}.
+         * @param requestOptions The requestOptions parameter for the {@code operation}.
+         * @return chat completions for the provided chat messages.
+         */
+        Mono<ChatCompletions> invoke(BinaryData completeRequest, RequestOptions requestOptions);
+    }
+
+    /**
+     * Reference to the async operation performing the actual completion streaming call.
+     */
+    @FunctionalInterface
+    public interface StreamingCompleteOperation {
+        /**
+         * invokes the operation.
+         *
+         * @param completeRequest The completeRequest parameter for the {@code operation}.
+         * @param requestOptions The requestOptions parameter for the {@code operation}.
+         * @return chat completions streaming for the provided chat messages.
+         */
+        Flux<StreamingChatCompletionsUpdate> invoke(BinaryData completeRequest, RequestOptions requestOptions);
+    }
+    //</editor-fold>
+
+    /**
+     * Creates ChatCompletionClientTracer.
+     *
+     * @param endpoint the service endpoint.
+     */
+    ChatCompletionClientTracer(String endpoint) {
+        this(endpoint, null, null);
```

Comment:
I noticed we call this one when building a client, so config and options are ignored.

I sent https://github.com/anuchandy/azure-sdk-for-java/pull/18 to fix it. It moves `ChatCompletionClientTracer` creation to client builder allowing to:
- pass properly configured tracer to the HttpPipeline (it's used by http tracing policy)
- create it with config and client options
- pass the whole `ChatCompletionClientTracer` instead of creating it in the client itself - this way we can inject configured one if necessary for tracing and also don't need to pass around all the parameters it need for initialization

I think we need to add at least one happy-case live test so we can  cover the end-to-end thing and catch things like this.

Reply:
Thank you for the pr!, It looks awesome, and I've merged the commits. (I made few cosmetic tweaks in `ChatCompletionsClientBuilder` so the "Verify TypeSpec Code Generation" CI stage (the diff validation b/w autogen & manual updates) passes.)

Reply:
Resolving other feedback already addressed in the merged commits.

Reply:
added live test as well.

Reply:
thank you!!